src/cogent3/core/new_alignment.py

```diff
@@ -3096,174 +3315,881 @@ def gapped_bytes_value(self) -> bytes:
         )
 
     def __str__(self) -> str:
-        # refactor: design
-        # should this return gapped or ungapped?
-        return self.gapped_str_value
+        return self.str_value
 
     def __array__(self, dtype=None, copy=None) -> numpy.ndarray:
-        arr = self.gapped_array_value
+        arr = self.array_value
         if dtype:
             arr = arr.astype(dtype)
         return arr
 
     def __bytes__(self) -> bytes:
-        return self.gapped_bytes_value
+        return self.bytes_value
+
+    def __getitem__(self, segment) -> new_sequence.SeqViewABC:
```

Comment:
Shouldn't the return type should be the aligned seq view abstract base class?

Reply:
indeed it should!